SourceCode/AgOpenGPS.Core/Streamers/Field/RecordedPathStreamer.cs

```diff
@@ -0,0 +1,97 @@
+﻿using AgOpenGPS.Core.Interfaces;
+using AgOpenGPS.Core.Models;
+using System;
+using System.Globalization;
+using System.IO;
+
+namespace AgOpenGPS.Core.Streamers
+{
+    public class RecordedPathStreamer : FieldAspectStreamer
+    {
+        public RecordedPathStreamer(
+            ILogger logger
+        )
+            : base(logger, "RecPath.txt")
+        {
+        }
+
+        public RecordedPath TryRead(string fieldPath, string fileName)
+        {
+            RecordedPath recordedPath = null;
+            try
+            {
+                recordedPath = Read(fieldPath, fileName);
+            }
+            catch (Exception e)
+            {
+                _presenter.PresentRecordedPathFileCorrupt();
+                _logger.LogError("Load Recorded Path" + e.ToString());
+            }
+            return recordedPath;
+        }
+
+        public void TryWrite(RecordedPath recordedPath, string fieldPath, string fileName)
+        {
+            CreateDirectory(fieldPath);
+            Write(recordedPath, fieldPath, fileName);
+        }
+
+        private RecordedPath Read(string fieldPath, string fileName)
+        {
+            RecordedPath recordedPath = new RecordedPath();
+            string fn = fileName != null ? fileName : _defaultFileName;
+            string fileAndDirectory = fieldPath + "\\" + fn;
+            if (File.Exists(fileAndDirectory))
```

Comment:
Invert the `if` and do an early return.
```suggestion
            if (!File.Exists(fileAndDirectory))
                return recordedPath;
```

Reply:
I think it is the wrong way around now, isn't it?

Reply:
Yes, a bug. Will change TramLines similarly.